### DISCUSSION TITLE
Find JWT info in API Logs

### DISCUSSION THREAD

**Discussion by zernonia** [supabase/supabase#6380] [Questions] [answered]
I was hoping to find the jwt information in the edge_log, but I can't seems to find it anywhere. Anyone has any idea how could I retrieve jwt information from the logs?

(headers doesn't shows any Bearer token info)
![image](https://user-images.githubusercontent.com/59365435/162574546-2d4dd289-9ace-4e97-894c-fcc5023d72c4.png)

**Comment by FelixZY** [comment, marked as answer]
I'm looking for this as well. Perhaps this has been excluded as the JWT can be considered a credential? In that case it would be really nice if at least the payload of the JWT was included.

For example: I'm minting my own JWT's and assign them an `id`. I would really like to know which requests are associated with a given JWT. If the JWT payload was included, it would be possible to at least infer this information using the `id` property.

**Reply by J0** [reply]
Hey @FelixZY and @zernonia,

Thanks for your patience so far -- chanced upon this.

You are right --  the JWT is excluded from the logs as it is considered a credential. Unfortunately, we don't support the notion of requests associated with a given JWT at the moment like you'd get in a session but it is something we are looking into.  We will take this feedback into account as we work on additional features -- do let us know if you have any further questions or concerns as well.

Thanks!

**Reply by FelixZY** [reply]
@J0 I'm currently working on minting JWTs to allow for the generation of API keys. I have found a stable way to do this from within PostgreSQL and will be posting my findings in a few related threads in a couple of days (https://github.com/supabase/supabase/issues/7186, https://github.com/supabase/supabase/discussions/4419).

However, one of the things I wish to implement is request limiting based on past usage. In this I find myself blocked by the ability to read request logs from within Postgres (https://github.com/supabase/supabase/discussions/8175) and the ability to identify the JWT in use (this issue). My current workaround is costly and involves an http request from within Postgres which would preferably be avoided.

In the future, I hope an API key generation feature will be supported out of the box by supabase. I also think the solution I'm working on is good enough to serve as a base for such a feature. If you would be interested, I would love to share my findings and discuss my challenges with your team.
